mobile/src/main/java/rs/readahead/washington/mobile/views/fragment/uwazi/entry/UwaziSelectEntitiesFragment.kt

```diff
@@ -0,0 +1,148 @@
+        initializeData()
+        initViews()
+    }
+
+    private fun initializeData() {
+        if (arguments?.getString(UWAZI_TEMPLATE) != null) {
+            arguments?.getString(UWAZI_TEMPLATE)?.let {
+                uwaziParser.parseTemplateForRelationShipEntities(it)
+            }
+        }
+
+        if (arguments?.getString(UWAZI_ENTRY_PROMPT_ID) != null) {
+            val id = arguments?.getString(UWAZI_ENTRY_PROMPT_ID)
+            val property =
+                uwaziParser.getTemplate()?.entityRow?.properties?.find { property -> property._id == id }
+            var i = 0
+            property?.entities?.forEach { value ->
+                items.add(SearchableItem(value.label, value.id, i.toString()))
+                i++
+            }
+        }
+
+        if (arguments?.getString(UWAZI_SELECTED_ENTITIES) != "[]") {
+            val relationShipEntities = arguments?.getString(UWAZI_SELECTED_ENTITIES)
+            val type = object : TypeToken<List<UwaziRelationShipEntity>>() {}.type
+            val listEntities: List<UwaziRelationShipEntity> =
+                Gson().fromJson(relationShipEntities, type)
+            if (listEntities.isNotEmpty()) {
+                listEntities.forEach { entity ->
+                    items.forEach { item ->
+                        if (entity.value == item.value)
+                            item.isSelected = true
+                    }
+                }
+
+            }
+        }
+    }
+
+    private fun initViews() {
+        val txtSearch: EditText =
+            binding.searchView.findViewById(androidx.appcompat.R.id.search_src_text)
+        txtSearch.setHintTextColor(Color.WHITE)
+        txtSearch.setTextColor(Color.WHITE)
+
+        val mLayoutManager = LinearLayoutManager(context)
+        binding.recyclerView.apply {
+            itemAnimator = null
+            layoutManager = mLayoutManager
+            adapter = setupAdapter()
+        }
+        binding.searchView.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
+            override fun onQueryTextSubmit(query: String) = false
+
+            override fun onQueryTextChange(newText: String): Boolean {
+                (binding.recyclerView.adapter as? SearchableAdapter)?.filter?.filter(newText)
+                return false
+            }
+        })
+
+        binding.searchView.setOnQueryTextFocusChangeListener { _, hasFocus ->
+            binding.searchView.isSelected = hasFocus
+            binding.searchView.isIconified = !hasFocus
+        }
+
+        binding.toolbar.apply {
+            setRightIcon(R.drawable.ic_check_white)
+            setRightIconVisibility(false)
+            onRightClickListener = {
+                (items.filter { it.isSelected }
+                    .toMutableList() as ArrayList<SearchableItem>).forEach {
+                    resultList.add(
+                        UwaziRelationShipEntity(it.value, it.text)
+                    )
```

Comment:
Let's Simplify the filter and mapping operator -> ` items.filter { it.isSelected }.map {`